desc/compute/utils.py

```diff
@@ -134,9 +146,14 @@ def get_data_deps(keys):
     keys = [keys] if isinstance(keys, str) else keys
 
     def _get_deps_1_key(key):
-        if "full_dependencies" in data_index[key]:
+        if has_axis:
+            if "full_with_axis_dependencies" in data_index[key]:
+                return data_index[key]["full_with_axis_dependencies"]["data"]
+        elif "full_dependencies" in data_index[key]:
             return data_index[key]["full_dependencies"]["data"]
         deps = data_index[key]["dependencies"]["data"]
+        if has_axis:
+            deps += data_index[key]["dependencies"]["limit_data"]
         if len(deps) == 0:
             return deps
         out = deps.copy()
```

Comment:
curious why we did this

Reply:
It's to avoid modifying the actual data index, since `deps = data_index[key]["dependencies"]["data"]` is just a pointer to the list. I think you might need the same thing on line 156.

```python
a = {"b": [1,2,3]}
c = a['b']
c += [4,5,6]

print(a)
>>> {'b': [1, 2, 3, 4, 5, 6]}

Reply:
Ahh this I need to correct then. The logic here had assumed `deps` was already a copy.  I guess I got used to working with jax arrays and was primed to think a new copy is always created on modification.